gno.land/pkg/integration/testing_integration.go

```diff
@@ -211,10 +244,42 @@ func setupGnolandTestScript(t *testing.T, txtarDir string) testscript.Params {
 
 				tsValidateError(ts, "gnokey", neg, err)
 			},
+			// adduser commands must be executed before starting the node; it errors out otherwise.
+			"adduser": func(ts *testscript.TestScript, neg bool, args []string) {
+				if nodeIsRunning(nodes, getNodeSID(ts)) {
+					tsValidateError(ts, "adduser", neg, errors.New("adduser must be used before starting node"))
+					return
+				}
+
+				if len(args) == 0 {
+					ts.Fatalf("new user name required")
+				}
+
+				kb, err := keys.NewKeyBaseFromDir(gnoHomeDir)
+				if err != nil {
+					ts.Fatalf("unable to get keybase")
+				}
+
+				balance, err := createAccount(ts, kb, args[0])
+				if err != nil {
+					ts.Fatalf("error creating account %s: %s", args[0], err)
+				}
+
+				newUserBalances = append(newUserBalances, balance)
+			},
 		},
 	}
 }
 
+func getNodeSID(ts *testscript.TestScript) string {
+	return ts.Getenv("SID")
+}
+
+func nodeIsRunning(nodes map[string]*testNode, sid string) bool {
```

Comment:
What do you think about defining a type for `map[string]*testNode`, and having this be a method you can call?

Reply:
I thought about it but I don't think it's worth the effort to create a new type to call one method.